ui/app/AppLayouts/Wallet/popups/swap/SwapSignModal.qml

```diff
@@ -0,0 +1,251 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQml.Models 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1 as SQUtils
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+import AppLayouts.Wallet.panels 1.0
+import AppLayouts.Wallet.popups 1.0
+
+import utils 1.0
+
+SignTransactionModalBase {
+    id: root
+
+    required property string fromTokenSymbol
+    required property string fromTokenAmount
+    required property string fromTokenContractAddress
+
+    required property string toTokenSymbol
+    required property string toTokenAmount
+    required property string toTokenContractAddress
+
+    required property string accountName
+    required property string accountAddress
+    required property string accountEmoji
+    required property string accountColorId
+
+    required property string networkShortName // e.g. "oeth"
+    required property string networkName // e.g. "Optimism"
+    required property string networkIconPath // e.g. `Style.svg("network/Network=Optimism")`
+    required property string networkBlockExplorerUrl
+
+    required property string currentCurrency
+    required property string fiatFees
+    required property string cryptoFees
+    required property double slippage
+
+    required property int loginType // RootStore.loginType -> Constants.LoginType enum
+
+    property bool loading // refreshing fees
+
+    property string serviceProviderName: "Paraswap"
+    property string serviceProviderURL: "https://www.paraswap.io/"
+
+    title: qsTr("Sign Swap")
+    //: e.g. (swap) 100 DAI to 100 USDT
+    subtitle: qsTr("%1 %2 to %3 %4").arg(formatBigNumber(fromTokenAmount)).arg(fromTokenSymbol)
+        .arg(formatBigNumber(toTokenAmount)).arg(toTokenSymbol)
+
+    gradientColor: Utils.setColorAlpha(d.walletColor, 0.05) // 5% of wallet color
+    fromImageSource: Constants.tokenIcon(root.fromTokenSymbol)
+    toImageSource: Constants.tokenIcon(root.toTokenSymbol)
+
+    //: e.g. "Swap 100 DAI to 100 USDT in <account name> on <network chain name>"
+    headerMainText: qsTr("Swap %1 %2 to %3 %4 in %5 on %6").arg(formatBigNumber(root.fromTokenAmount)).arg(root.fromTokenSymbol)
+        .arg(formatBigNumber(root.toTokenAmount)).arg(root.toTokenSymbol).arg(root.accountName).arg(root.networkName)
+    headerSubTextLayout: [
+        StatusBaseText {
+            font.pixelSize: Style.current.additionalTextSize
+            text: qsTr("Powered by")
+            onLinkActivated: d.openLinkWithConfirmation(root.serviceProviderURL)
```

Comment:
probably not needed

Reply:
what is not needed?

Reply:
`onLinkActivated: d.openLinkWithConfirmation(root.serviceProviderURL)` on the `Powered by` text

Reply:
Ah, got it! :)

Reply:
Removed